tests/manage/mcg/conftest.py

```diff
@@ -104,6 +111,189 @@ def bucket_cleanup():
     return _create_buckets
 
 
+@pytest.fixture(scope='class')
+def cloud_uls_factory(request, cld_mgr):
+    """
+        Create a Underlying Storage factory.
+        Calling this fixture creates a new underlying storage(s).
+        Args:
+            cld_mgr (CloudManager): Cloud Manager object containing all connections to clouds
+    """
+    all_created_uls = {
+        'aws': set(),
+        'google': set(),
+        'azure': set(),
+        's3comp': set()
+    }
+
+    ulsMap = {
+        'aws': cld_mgr.aws_client,
+        'google': cld_mgr.google_client,
+        'azure': cld_mgr.azure_client,
+        's3comp': cld_mgr.s3comp_client
+    }
+
+    def _create_uls(uls_dict):
+        """
+        Creates and deletes all underlying storage that were created as part of the test
+        Args:
+            uls_dict (dict): Dictionary containing storage provider as key and a list of tuples
+            as value.
+            each tuple contain amount as first parameter and region as second parameter.
+            Example:
+                'aws': [(3,us-west-1),(2,eu-west-2)]
+        Returns:
+            dict: A dictionary of cloud names as keys and uls names sets as value.
+        """
+        current_call_created_uls = {
+            'aws': set(),
+            'google': set(),
+            'azure': set(),
+            's3comp': set()
+        }
+
+        for cloud, params in uls_dict.items():
+            if cloud.lower() not in ulsMap:
+                raise RuntimeError(
+                    f'Invalid interface type received: {cloud}. '
+                    f'available types: {", ".join(ulsMap.keys())}'
+                )
+            for tup in params:
+                amount, region = tup
+                for i in range(amount):
+                    uls_name = create_unique_resource_name(
+                        resource_description='uls', resource_type=cloud.lower()
+                    )
+                    ulsMap[cloud.lower()].create_uls(uls_name, region)
+                    all_created_uls[cloud].add(uls_name)
+                    current_call_created_uls[cloud.lower()].add(uls_name)
+
+            return current_call_created_uls
+
+    def uls_cleanup():
+        for cloud, uls_set in all_created_uls.items():
+            client = ulsMap[cloud]
+            if client is not None:
+                all_existing_uls = client.get_all_uls_names()
+                for uls in uls_set:
+                    if uls in all_existing_uls:
+                        logger.info(f'Cleaning up uls {uls}')
+                        client.delete_uls(uls)
+                        logger.info(
+                            f"Verifying whether uls: {uls} exists after deletion"
+                        )
+                        sleep(5)  # wait for aws command to pass
```

Comment:
is there anything we can poll here instead of just sleep? there is a chance that in some cases 5 seconds will not be enough

Reply:
removed